sequencer/src/bin/espresso-dev-node.rs

```diff
@@ -0,0 +1,332 @@
+
+    let sequencer_url =
+        Url::parse(format!("http://localhost:{}", opt.sequencer_api_port).as_str()).unwrap();
+    let commitment_task_options = CommitmentTaskOptions {
+        l1_provider: url,
+        l1_chain_id: None,
+        hotshot_address,
+        sequencer_mnemonic: opt.mnemonic,
+        sequencer_account_index: opt.account_index,
+        query_service_url: Some(sequencer_url),
+        request_timeout: Duration::from_secs(5),
+        delay: None,
+    };
+
+    tracing::info!("starting hotshot commitment task");
+    run_hotshot_commitment_task::<es_version::SequencerVersion>(&commitment_task_options).await;
+
+    Ok(())
+}
+
+// Copied from `commitment_task::start_http_server`.
+// TODO: Remove these redundant code
+fn start_commitment_server<Ver: StaticVersionType + 'static>(
+    port: u16,
+    hotshot_address: Address,
+    bind_version: Ver,
+) -> io::Result<()> {
+    let mut app = tide_disco::App::<(), ServerError>::with_state(());
+    let toml = toml::from_str::<toml::value::Value>(include_str!("../../api/commitment_task.toml"))
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    let mut api = Api::<(), ServerError, Ver>::new(toml)
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    api.get("gethotshotcontract", move |_, _| {
+        async move { Ok(hotshot_address) }.boxed()
+    })
+    .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    app.register_module("api", api)
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    spawn(app.serve(format!("0.0.0.0:{port}"), bind_version));
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{process::Stdio, time::Duration};
+
+    use async_compatibility_layer::{
+        art::async_sleep,
+        logging::{setup_backtrace, setup_logging},
+    };
+    use async_std::process::Command;
+    use committable::{Commitment, Committable};
+    use es_version::SequencerVersion;
+    use escargot::CargoBuild;
+    use portpicker::pick_unused_port;
+    use reqwest::StatusCode;
+    use sequencer::Transaction;
+    use sequencer_utils::AnvilOptions;
+
+    // If this test failed and you are doing changes on the following stuff, please
+    // sync your changes to [`espresso-sequencer-go`](https://github.com/EspressoSystems/espresso-sequencer-go)
+    // and open a PR.
+    // - APIs update
+    // - Types (like `Header`) update
+    #[async_std::test]
+    async fn dev_node_test() {
+        setup_logging();
+        setup_backtrace();
+        let anvil = AnvilOptions::default().spawn().await;
+        // let commitment_task_port = pick_unused_port().unwrap();
+        let commitment_task_port = 10002;
+        let api_port = pick_unused_port().unwrap();
+        let postgres_port = pick_unused_port().unwrap();
+
+        let mut db = Command::new("docker")
+            .arg("compose")
```

Comment:
Let's not introduce a dependency on docker compose and call docker directly instead.

Reply:
This will be replaced by TmpDb from query service

Reply:
Done in https://github.com/EspressoSystems/espresso-sequencer/pull/1353/commits/76dec3068f2d29c1a582a5693a3a1eae4e26cc06